crates/miden-tx/src/executor/notes_checker.rs

```diff
@@ -74,30 +78,35 @@ where
     /// This function attempts to find the maximum set of notes that can be successfully executed
     /// together by the target account.
     ///
-    /// If some notes succeed but others fail, the failed notes are removed from the candidate set
+    /// Because of the runtime complexity involved in this function, a limited range of
+    /// [`Self::INPUT_NOTES_RANGE`] input notes is allowed.
+    ///
+    /// If some notes succeed and others fail, the failed notes are removed from the candidate set
     /// and the remaining notes (successful + unattempted) are retried in the next iteration. This
     /// process continues until either all remaining notes succeed or no notes can be successfully
     /// executed
     ///
-    /// # Example Execution Flow
-    ///
-    /// Given notes A, B, C, D, E:
+    /// For example, given notes A, B, C, D, E, the execution flow would be as follows:
     /// - Try [A, B, C, D, E] → A, B succeed, C fails → Remove C, try again.
     /// - Try [A, B, D, E] → A, B, D succeed, E fails → Remove E, try again.
     /// - Try [A, B, D] → All succeed → Return successful=[A, B, D], failed=[C, E].
     ///
-    /// # Returns
+    /// If a failure occurs at the epilogue phase of the transaction execution, the relevant set of
+    /// otherwise-successful notes are retried in various combinations in an attempt to find a
+    /// combination that passes the epilogue phase successfully.
```

Comment:
Should we describe the approach in more detail to give callers an idea of the used algorithm, or is that unnecessary?